### DISCUSSION TITLE
Restoring a SQL Server Database Backup using .NET Aspire

### DISCUSSION THREAD

**Discussion by bytefish** [microsoft/aspire#4390] [Q&A] [answered]
I would like to update my WideWorldImporters sample application to Aspire:

* https://github.com/bytefish/WideWorldImporters

I think it could be a nice showcase for .NET Aspire, because it contains a Backend, Frontend and a non-trivial SQL Server database. And  I want users to press F5 and experiment right away, instead of fiddling with Docker and all that.

What I am unsure about is, what's the best way to download and restore the database backup, which can be obtained from the SQL Server Examples repositories (https://github.com/Microsoft/sql-server-samples/releases). When I wrote the `Dockerfile` for the application, it took me quite some time to figure out, that the `sqlcmd` coming with the `mssql/server:2022-latest` image hangs, when restoring full backups. This is unrelated to Aspire, but it would be nice to know, if it's somehow possible to add the `mssql-tools18` package to the image, so the restore works correctly.

I am currently downloading the Backup it in a `Dockerfile` like this:

```dockerfile
# ...

RUN apt-get update \
    && ACCEPT_EULA=Y apt-get install mssql-tools18 unixodbc-dev -y \
    && wget -q  https://github.com/Microsoft/sql-server-samples/releases/download/wide-world-importers-v1.0/WideWorldImporters-Full.bak

# ...
```

And restore it using an SQL Script, which is called by the `entrypoint.sh`:

* https://github.com/bytefish/WideWorldImporters/blob/main/docker/db/restore.sql

I saw the DatabaseContainers example looks strikingly similar:

* https://github.com/dotnet/aspire-samples/tree/main/samples/DatabaseContainers/DatabaseContainers.AppHost

Are there any more pointers or best practices for restoring SQL Server Database Backups using Aspire?

Or is it better to wait until Dockerfile support is available, which has been described in:

* https://github.com/dotnet/aspire/issues/1852

**Comment by mitchdenny** [comment, marked as answer]
We are adding support for building a container from a Dockerfile and running it. Once we have this you'll have a fairly concise way to inject scripty kinds of things like this into your application model. We would build a container image based on the sql tools image which includes your script, your script would download the database and restore it and then exit.

**Reply by mitchdenny** [reply]
Check out this PR for our progress #dockerfile: https://github.com/dotnet/aspire/pull/4312

**Reply by bytefish** [reply]
Thanks, I'll just sit it out until Dockerfiles are supported. 🫣

@mitchdenny Great work and discussions in the PR by the way! 👍

**Reply by bytefish** [reply]
@mitchdenny Is there a timeline, when it’s going to be released? Maybe I have missed the planned Release cycle.

**Reply by mitchdenny** [reply]
We are hoping to release this month. It is a content driven release so it'll ship when what we wanted to put in 8.1 is ready.
